libsql/internal/hrana/batch_result_test.go

```diff
@@ -41,7 +42,7 @@ func TestBatchResult_UnmarshalJSON(t *testing.T) {
 	for i, tc := range testCases {
 		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
 			batchResult := &BatchResult{}
-			err := batchResult.UnmarshalJSON(tc.jsonData)
+			err := json.Unmarshal(tc.jsonData, batchResult)
```

Comment:
We always use `json.Unmarshal` never the `struct.Unmarshal` even though it is same. Fixed it here in this comment.